indexer/services/vulcan/src/handlers/order-replace-handler.ts

```diff
@@ -200,9 +274,114 @@ export class OrderReplaceHandler extends Handler {
       oldOrderId,
       removeOrderResult,
     });
+
+    const stateRemainingQuantums: Big = await getStateRemainingQuantums(
+      removeOrderResult.removedOrder!,
+    );
+
+    // If the remaining amount of the order in state is <= 0, the order is filled and
+    // does not need to have it's status updated
+    let canceledOrder: OrderFromDatabase | undefined;
+    if (stateRemainingQuantums.gt(0)) {
+      canceledOrder = await runFuncWithTimingStat(
+        this.cancelOrderInPostgres(oldOrderId),
+        this.generateTimingStatsOptions('cancel_order_in_postgres'),
+      );
+    } else {
+      canceledOrder = await runFuncWithTimingStat(
+        OrderTable.findById(OrderTable.orderIdToUuid(oldOrderId)),
+        this.generateTimingStatsOptions('find_order'),
+      );
+    }
+
+    const subaccountMessage: Message = {
+      value: this.createSubaccountWebsocketMessageFromRemoveOrderResult(
+        removeOrderResult,
+        canceledOrder,
+        oldOrderId,
+        perpetualMarket,
+        blockHeightRefresher.getLatestBlockHeight(),
+      ),
+      headers,
+    };
+
+    sendMessageWrapper(subaccountMessage, KafkaTopics.TO_WEBSOCKETS_SUBACCOUNTS);
+
     return removeOrderResult;
   }
 
+  // eslint-disable-next-line @typescript-eslint/require-await
+  protected async cancelOrderInPostgres(
+    oldOrderId: IndexerOrderId,
+  ): Promise<OrderFromDatabase | undefined> {
+    return OrderTable.update({
+      id: OrderTable.orderIdToUuid(oldOrderId),
+      status: OrderStatus.CANCELED,
+    });
+  }
+
+  protected createSubaccountWebsocketMessageFromRemoveOrderResult(
```

Comment:
is this used in order-remove-handler as well? can we consolidate? feel free to tackle in a separate PR

Reply:
I made some slight changes to it because the order replace message doesn't contain an order removal reason, so i hardcode it to `OrderRemovalReason.ORDER_REMOVAL_REASON_USER_CANCELED` in the order replace handler, which matches the old behavior of canceling and placing a new order